src/webgpu/api/operation/vertex_state.spec.ts

```diff
@@ -0,0 +1,348 @@
+export const description = '';
+
+import { params, poptions } from '../../../common/framework/params_builder.js';
+import { makeTestGroup } from '../../../common/framework/test_group.js';
+import { EncodableTextureFormat } from '../../capability_info.js';
+import { GPUTest } from '../../gpu_test.js';
+import { getTextureCopyLayout } from '../../util/texture/layout.js';
+
+const kHeight = 4;
+const kWidth = 4;
+const kTextureFormat: GPUTextureFormat = 'bgra8unorm';
+
+/**
+ * The expected rendering shapes:
+ * Square:
+ *  0---------3
+ *  |         |
+ *  |         |
+ *  |         |
+ *  2---------1
+ * BottomLeftTriangle:
+ *  0
+ *  | \
+ *  |    \
+ *  |       \
+ *  2---------1
+ * Points:
+ *  0         3
+ *
+ *
+ *
+ *  2         1
+ * XShape:
+ *  0         3
+ *    \     /
+ *       \
+ *    /     \
+ *  2         1
+ */
+const enum RenderShape {
+  Square = 'Square',
+  BottomLeftTriangle = 'BottomLeftTriangle',
+  Points = 'Points',
+  XShape = 'XShape',
+  Nothing = 'Nothing',
+}
+
+class IndexFormatTest extends GPUTest {
+  private colorAttachment!: GPUTexture;
+  byteLength!: number;
+  rowsPerImage!: number;
+  bytesPerRow!: number;
+  result!: GPUBuffer;
+
+  async init(): Promise<void> {
```

Comment:
Is it necessary to do this work in init() and store into class members, instead of just as a helper called in `run()`? I think it looks like only `result` is needed, which could just be returned from `run()`.